server/utils/profileRankingUtils.js

```diff
@@ -52,17 +56,37 @@ const evaluateCandidate = async (user, candidate) => {
         popularityOvr = totalPopularity / candidatePosts.length;
     }
 
```

Comment:
it's getting a bit hard to read this function. It'd help if you modularized code a bit by extracting some of the different steps you are taking here into functions. 

For example, you could have a function called `getSimilarityFactor` that takes `user` and `candidate` as params, does all the fetching, and returns the similarityFactor. You could do a similar thing for other factors you are using